core/src/org/javarosa/core/model/condition/EvaluationContext.java

```diff
@@ -199,13 +202,21 @@ public Object getVariable(String name) {
         return expandReference(ref, false);
     }
 
-    // take in a potentially-ambiguous ref, and return a vector of refs for all nodes that match the passed-in ref
-    // meaning, search out all repeated nodes that match the pattern of the passed-in ref
-    // every ref in the returned vector will be unambiguous (no index will ever be INDEX_UNBOUND)
-    // does not return template nodes when matching INDEX_UNBOUND, but will match templates when INDEX_TEMPLATE is explicitly set
-    // return null if ref is relative, otherwise return vector of refs (but vector will be empty is no refs match)
-    // '/' returns {'/'}
-    // can handle sub-repetitions (e.g., {/a[1]/b[1], /a[1]/b[2], /a[2]/b[1]})
+    /**
+     * Search for all repeated nodes that match the pattern of the 'ref'
+     * argument.
+     *
+     * '/' returns {'/'} 
+     * can handle sub-repetitions (e.g., {/a[1]/b[1], /a[1]/b[2], /a[2]/b[1]})
+     *
+     * @param ref Potentially ambiguous reference
+     * @param includeTemplates
+     * @return Null if 'ref' is relative reference. Otherwise, returns a vector
+     * of references that point to nodes that match 'ref' argument. These
+     * references are unambiguous (no index will ever be INDEX_UNBOUND) template
+     * nodes won't be included when matching INDEX_UNBOUND, but will be when
+     * INDEX_TEMPLATE is explicitly set.
```

Comment:
This comment doesn't really make much sense, should INDEX_TEMPLATE be replaced by includeTemplates in the doc?